src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -507,36 +511,51 @@ bool is_numeric_property_value_superset(
   json::Value const& older,
   json::Value const& newer,
   std::string_view prop_name,
-  VPred&& value_predicate) {
-    auto [maybe_is_compatible, older_val_p, newer_val_p]
-      = extract_property_and_gate_check(older, newer, prop_name);
-    if (maybe_is_compatible.has_value()) {
-        return maybe_is_compatible.value();
+  VPred&& value_predicate,
+  std::optional<double> default_value = std::nullopt) {
+    // get value or default_value
+    auto get_value = [&](json::Value const& v) -> std::optional<double> {
```

Comment:
I think this only fixes the bug for the numeric properties but the same bug exists for other calls to `extract_property_and_gate_check`. If I understand it correctly, the bug is that `extract_property_and_gate_check` doesn't handle the default value of the field.
For example, this test case would fail - I believe incorrectly:
```
{
    .reader_schema
    = R"({"type": "number", "minimum": 11, "exclusiveMinimum": false})",
    .writer_schema = R"({"type": "number", "minimum": 11})",
    .reader_is_compatible_with_writer = true,
  },
```

Reply:
You are right, but `extract_property_and_gate_check` cannot handle this case cleanly. 
"exclusiveMinimum" (or _maximum) is kind of tricky because it's a boolean with a default value in draft4 but a number without a default value from draft6.
Currently, `is_numeric_superset` doesn't know the schema dialect, so it cannot assign a meaningful default to the property. 
We have json schema normalization on the roadmap that should fix this problem at the root, by explicitly setting all the implicit default values based on the schema dialect in use.
So if you are ok with this, i'd move this fix to a follow-up or to the `json schema normalization` PR